libs/vkd3d/resource.c

```diff
@@ -2133,14 +2132,8 @@ static ULONG STDMETHODCALLTYPE d3d12_resource_Release(d3d12_resource_iface *ifac
     TRACE("%p decreasing refcount to %u.\n", resource, refcount);
 
     if (!refcount)
-    {
-        struct d3d12_device *device = resource->device;
-
         d3d12_resource_decref(resource);
 
-        d3d12_device_release(device);
```

Comment:
Do resources not take device references?

Reply:
We remove the device reference in `d3d12_resource_destroy` instead.